sklearn/preprocessing/_encoders.py

```diff
@@ -38,44 +38,65 @@ def _check_X(self, X):
         - convert list of strings to object dtype
         - check for missing values for object dtype data (check_array does
           not do that)
+        - return list of features (arrays)
```

Comment:
Can you add here that this list of features is constructed feature by feature to preserver the data types of pandas DataFrame columns ?